week-1-ts-ecosystem/w01e04-light-me-up/src/lightBulb.ts

```diff
@@ -0,0 +1,18 @@
+ import { PowerSource } from "./powerSource";
+
+export class LightBulb {
+  protected readonly powerConsumption = 20;
+
+  constructor(private powerSource: PowerSource) {}
+
+  turnOn() {
+    this.powerSource.consume(this.powerConsumption);
+    console.log("🔦");
+  }
+
+  turnOnDuring(seconds: number) {
+    for (let second = 0; second < seconds; second++) {
+      this.turnOn();
+    }
```

Comment:
powiesz mi że się czepiam ale to nie jest asynchroniczne :) 
poproszę o jakiś `setTimeout` i poprawne działanie.

Reply:
ale sama koncepcja w dechę - wszystko w porządku jak najbardziej.

Reply:
Taka Twoja rola, żeby się czepiać 🧑‍🏫 
Dodałem setTimeout w tej metodzie. Teraz 🔦 pojawia się co sekundę, a w 6 wyrzuca wyjątek.

Super, dzięki 🎉